core/src/main/scala/chisel3/probe/package.scala

```diff
@@ -68,10 +68,30 @@ package object probe extends SourceInfoDoc {
     }
   }
 
+  /** Pad [[Data]] to the width of a probe if it supports padding */
+  private def padDataToProbeWidth[T <: Data](data: T, probe: Data)(implicit sourceInfo: SourceInfo): T = {
+    // check probe width is known
+    requireHasProbeTypeModifier(probe, s"Expected $probe to be a probe.")
+    if (!probe.isWidthKnown) Builder.error("Probe width unknown.")
+    val probeWidth = probe.widthOption.getOrElse(0)
+
+    // check data width is known
+    data.widthOption match {
+      case None => Builder.error("Data width unknown.")
+      case Some(w) =>
+        if (w > probeWidth) Builder.error(s"Data width $w is larger than $probeWidth.")
```

Comment:
If probe width is unknown, but data width is known, there's an extra error being generated that is unnecessary. I think you should tweak this like so:
```suggestion
      case Some(w) =>
        if (probe.widthOption.exists(w > _)) Builder.error(s"Data width $w is larger than $probeWidth.")
```